inspirehep/modules/workflows/tasks/actions.py

```diff
@@ -217,11 +216,19 @@ def is_experimental_paper(obj, eng):
 @with_debug_logging
 def is_arxiv_paper(obj, *args, **kwargs):
     """Check if the record is from arXiv."""
-    arxiv_id = get_arxiv_id(obj.data)
-    categories = get_value(obj.data, 'arxiv_eprints.categories')
 
-    if arxiv_id or categories:
-        return True
+    acquisition_source = obj.data.get('acquisition_source')
+
+    if acquisition_source:
+        method = acquisition_source.get('method')
+
+        if method == 'submitter':
+            return 'arxiv_eprints' in obj.data
+
+        elif method == 'hepcrawl':
+            source = acquisition_source.get('source', '')
+            return source.lower() == 'arxiv'
+
```

Comment:
WDYT of:
```python
method = get_value(obj.data, 'acquisition_source.method')
source = get_value(obj.data, 'acquisition_source.source', default='')

is_submission_with_arxiv = method == 'submitter' and 'arxiv_eprints' in obj.data
is_harvested_from_arxiv = method == 'hepcrawl' and source.lower() == 'arxiv'

return is_submission_with_arxiv or is_harvested_from_arxiv
```
?

Reply:
Seems more readable thanks to variables naming, I can go with it.